dot/core/service.go

```diff
@@ -491,6 +493,10 @@ func (s *Service) ProcessBlockRequestMessage(msg network.Message) error {
 	// get sub-chain of block hashes
 	subchain := s.blockState.SubChain(startHash, endHash)
 
+	if len(subchain) > maxResponseSize {
+		subchain = subchain[:maxResponseSize]
+	}
```

Comment:
should we log a warning that we are trimming the subchain due to max response size?

Reply:
I don't think it needs a warning, it's not really something the sender of the block response would need to worry about